src/IO/WriteBufferFromS3.cpp

```diff
@@ -529,7 +529,9 @@ void WriteBufferFromS3::writePart(WriteBufferFromS3::PartData && data)
         return;
     }
 
+    /// all iterators are invalidated after push_back, so we create part_tag reference after push_back
```

Comment:
If you are writing about iterations invalidation, worth to write about that references are not invalidated in deque as well? Since this is important to make things works here

Also, this is the only reason for using deque here over vector?